kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -288,6 +304,60 @@ private BigQuery getBigQuery() {
     return bigQuery.updateAndGet(bq -> bq != null ? bq : newBigQuery());
   }
 
+  private void setTimePartitioningForTimestamp(PartitionedTableId.Builder builder, TimePartitioning timePartitioning,
+                                               Long timestamp) {
+    if (timePartitioning != null) {
+      switch (timePartitioning.getType()) {
+        case HOUR:
+          builder.setHourPartition(timestamp);
+          break;
+        case MONTH:
+          builder.setMonthPartition(timestamp);
+          break;
+        case YEAR:
+          builder.setYearPartition(timestamp);
+          break;
+        default:
+          builder.setDayPartition(timestamp);
+      }
+    } else {
+      builder.setDayPartition(timestamp);
+    }
+  }
+
+  private void setTimePartitioning(PartitionedTableId.Builder builder, TimePartitioning timePartitioning) {
+    if (timePartitioning != null) {
+      switch (timePartitioning.getType()) {
+        case HOUR:
+          builder.setHourPartitionNow();
+          break;
+        case MONTH:
+          builder.setMonthPartitionForNow();
+          break;
+        case YEAR:
+          builder.setYearPartitionForNow();
+          break;
+        default:
+          builder.setDayPartitionForNow();
+      }
+    } else {
+      builder.setDayPartitionForNow();
+    }
+  }
+
+  private Table retrieveCachedTable(TableId tableId) {
+    String key = tableId.getProject() + "." + tableId.getDataset() + "." + tableId.getTable();
+    Map<String, Table> cache = getCache();
+    Table table = cache.get(key);
+
+    if (table == null){
+      table = getBigQuery().getTable(tableId);
+      if (table != null) cache.put(key, table);
+    }
+
+    return table;
```

Comment:
I think we can simplify this section:
```suggestion
    return cache.computeIfAbsent(key, k -> getBigQuery().getTable(tableId));
```

Also, why use a custom string key here instead of the `TableId`?

Reply:
Reworked this piece of code to use map's computeIfAbsent method along with tableId as the key as opposed to customer key.